pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -473,6 +473,11 @@ private CompletableFuture<Long> individualAckNormal(CommandAck ack, Map<String,
             long ackedCount = 0;
             long batchSize = getBatchSize(msgId);
             Consumer ackOwnerConsumer = getAckOwnerConsumer(msgId.getLedgerId(), msgId.getEntryId());
+            if (Subscription.isIndividualAckMode(ackOwnerConsumer.subType())
+                && !ackOwnerConsumer.getPendingAcks().containsKey(msgId.getLedgerId(), msgId.getEntryId())) {
+                // this message has been acked, skip to keep stats correct
+                continue;
+            }
```

Comment:
The `ackOwnerConsumer` can be a different consumer. 

```java
private Consumer getAckOwnerConsumer(long ledgerId, long entryId) {
    Consumer ackOwnerConsumer = this;
    if (Subscription.isIndividualAckMode(subType)) {
        if (!getPendingAcks().containsKey(ledgerId, entryId)) {
            for (Consumer consumer : subscription.getConsumers()) {
                if (consumer != this && consumer.getPendingAcks().containsKey(ledgerId, entryId)) {
                    ackOwnerConsumer = consumer;
                    break;
                }
            }
        }
    }
    return ackOwnerConsumer;
}
```

We will still have the issue that the unacked messages count is decreased for this consumer, but actually the owner consumer is another consumer.

Reply:
I don't understand, if the actual consumer is another consumer, `addAndGetUnAckedMsgs` can update the unack count of that consumer:

https://github.com/apache/pulsar/blob/96fb7dab7a1bc260d8ee3c9d5df15de6863ee15b/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java#L1049-L1060